lib/autoload/send_file.rb

```diff
@@ -0,0 +1,20 @@
+module SendFile
+  # Makes the given file publicly accessible.
+  #
+  # @param [String] file The path to the local file.
+  # @param [String] folder_name The name of the working folder, this name should be unique.
+  # @param [String] file_name The name of public visible file.
+  # @return [String] The url to the public visible file, with the format of
+  #   "/downloads/#{folder_name}/#{file_name}"
+  def publish_file(file, folder_name, file_name)
```

Comment:
file: 'something.txt' => folder: '', file_name 'something.txt'

If a custom directory is needed then specify the file name explicitly? I also do not think there's a need to explicitly split file name from folder name.